gnovm/pkg/doc/json_doc.go

```diff
@@ -232,14 +384,65 @@ func (d *Documentable) extractValueSpecs(pkg *doc.Package, specs []ast.Spec) []*
 
 // mustFormatNode calls format.Node and returns the result as a string.
 // Panic on error, which shouldn't happen since the node is a valid AST from pkgData.parseFile.
-func mustFormatNode(fset *token.FileSet, node any) string {
+// If source is true and the optional ast.File is given, then use it to get internal comments.
+func mustFormatNode(fset *token.FileSet, node any, source bool, file *ast.File) string {
+	if !source {
+		// Omit the Doc and Body so that it's not in the signature
+		switch n := node.(type) {
+		case *ast.FuncDecl:
+			node = &ast.FuncDecl{
+				Recv: n.Recv,
+				Name: n.Name,
+				Type: n.Type,
+			}
+		case *ast.GenDecl:
+			node = &ast.GenDecl{
+				TokPos: n.TokPos,
+				Tok:    n.Tok,
+				Lparen: n.Lparen,
+				Specs:  n.Specs,
+				Rparen: n.Rparen,
+			}
+		}
+	}
+
+	if file != nil && source {
+		// Need an extra little dance to get internal comments to appear.
+		node = &printer.CommentedNode{
+			Node:     node,
+			Comments: file.Comments,
+		}
+	}
+
 	buf := new(strings.Builder)
 	if err := format.Node(buf, fset, node); err != nil {
 		panic("Error in format.Node: " + err.Error())
 	}
 	return buf.String()
 }
 
+// isCrossing returns true if the first param is "cur realm"
+func isCrossing(params []*JSONField) bool {
+	if len(params) < 1 {
+		return false
+	}
+
+	return params[0].Name == "cur" && params[0].Type == "realm"
```

Comment:
```suggestion
	return params[0].Type == "realm"
```

`cur` is not required, it could be another name